Installation issue: seacas@2023.05.30+faodel

### Steps to reproduce the issue

```console
$ spack spec -II seacas+mpi
Input spec
--------------------------------
 -   seacas@2023-05-30%gcc@11.3.0~adios2+applications+cgns+faodel+fortran+ipo+legacy+matio~metis+mpi+shared+thread_safe+x11 build_system=cmake build_type=RelWithDebInfo generator=make/6h6cnpfh

Concretized
--------------------------------
 -   6h6cnpf  seacas@2023-05-30%gcc@11.3.0~adios2+applications+cgns+faodel+fortran+ipo+legacy+matio~metis+mpi+shared+thread_safe+x11 build_system=cmake build_type=RelWithDebInfo generator=make arch=linux-rhel8-zen2
[+]  ocpnbir      ^cgns@4.3.0%gcc@11.3.0+base_scope+fortran+hdf5~int64+ipo+legacy~mem_debug+mpi+pic+scoping+shared+static+testing+tools build_system=cmake build_type=RelWithDebInfo generator=make arch=linux-rhel8-zen2
[+]  5ddypq7          ^hdf5@1.14.0%gcc@11.3.0+cxx+fortran+hl+ipo+java+map+mpi+shared+szip+threadsafe+tools api=default build_system=cmake build_type=RelWithDebInfo generator=make patches=0b5dd6f arch=linux-rhel8-zen2
[+]  sh24ggk              ^openjdk@11.0.17_8%gcc@11.3.0 build_system=generic arch=linux-rhel8-zen2
[+]  pho5cm4          ^libsm@1.2.3%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  efnu5dw              ^libice@1.0.9%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  jeqzbqg              ^libuuid@1.0.3%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  52ri3iu          ^libxmu@1.1.4%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  7vbnjnb              ^libxext@1.3.3%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  e26lecd              ^libxt@1.1.5%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  gyu3sh4          ^mesa-glu@9.0.2%gcc@11.3.0 build_system=autotools gl=osmesa arch=linux-rhel8-zen2
[+]  uqi43l7          ^osmesa@11.2.0%gcc@11.3.0 build_system=bundle arch=linux-rhel8-zen2
[+]  m5rhyzi              ^mesa@22.1.6%gcc@11.3.0+glx~llvm+opengl+opengles+osmesa~strip build_system=meson buildtype=debugoptimized default_library=shared,static patches=ee737d1 arch=linux-rhel8-zen2
[+]  edn5eua                  ^expat@2.5.0%gcc@11.3.0+libbsd build_system=autotools arch=linux-rhel8-zen2
[+]  czlbqsf                  ^glproto@1.4.17%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  4iuwrq7                  ^libunwind@1.6.2%gcc@11.3.0~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs+pic+tests+weak_backtrace+xz+zlib build_system=autotools components=coredump,ptrace,setjump libs=shared,static arch=linux-rhel8-zen2
[+]  na74xbv                  ^meson@1.1.0%gcc@11.3.0 build_system=python_pip patches=0f0b1bd arch=linux-rhel8-zen2
[+]  3gyusxe                  ^ninja@1.11.1%gcc@11.3.0+re2c build_system=generic arch=linux-rhel8-zen2
[+]  35p2wk5                      ^re2c@2.2%gcc@11.3.0 build_system=generic arch=linux-rhel8-zen2
[+]  7zzq2pb                  ^py-mako@1.2.2%gcc@11.3.0 build_system=python_pip arch=linux-rhel8-zen2
[+]  u75qdco                      ^py-markupsafe@2.1.1%gcc@11.3.0 build_system=python_pip arch=linux-rhel8-zen2
[+]  dzjjc4h                  ^xrandr@1.5.0%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  vqpyw2q                      ^libxrandr@1.5.3%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  fft4ssy                      ^randrproto@1.5.0%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  hj23ugq          ^tcl@8.6.12%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  btrblvx          ^tk@8.6.11%gcc@11.3.0+xft+xss build_system=autotools arch=linux-rhel8-zen2
[+]  r4gznpm              ^libxft@2.3.2%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  gxnigy4                  ^fontconfig@2.14.2%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  vxspw7h                      ^font-util@1.4.0%gcc@11.3.0 build_system=autotools fonts=encodings,font-adobe-100dpi,font-adobe-75dpi,font-adobe-utopia-100dpi,font-adobe-utopia-75dpi,font-adobe-utopia-type1,font-alias,font-arabic-misc,font-bh-100dpi,font-bh-75dpi,font-bh-lucidatypewriter-100dpi,font-bh-lucidatypewriter-75dpi,font-bh-type1,font-bitstream-100dpi,font-bitstream-75dpi,font-bitstream-speedo,font-bitstream-type1,font-cronyx-cyrillic,font-cursor-misc,font-daewoo-misc,font-dec-misc,font-ibm-type1,font-isas-misc,font-jis-misc,font-micro-misc,font-misc-cyrillic,font-misc-ethiopic,font-misc-meltho,font-misc-misc,font-mutt-misc,font-schumacher-misc,font-screen-cyrillic,font-sun-misc,font-winitzki-cyrillic,font-xfree86-type1 arch=linux-rhel8-zen2
[+]  vzqfxnm                          ^bdftopcf@1.1%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  vkwahrq                              ^fontsproto@2.1.3%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  7cdmzxd                              ^libxfont@1.5.4%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  qvrgzhq                          ^mkfontdir@1.0.7%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  glo64pt                          ^mkfontscale@1.2.2%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  hoqfdwc                              ^libfontenc@1.1.7%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  xsxnjdu                      ^gperf@3.1%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  4mvt3vj                  ^freetype@2.11.1%gcc@11.3.0+pic+shared build_system=autotools arch=linux-rhel8-zen2
[+]  4y72pm7                      ^libpng@1.6.39%gcc@11.3.0+ipo build_system=cmake build_type=RelWithDebInfo generator=make libs=shared,static arch=linux-rhel8-zen2
[+]  idhsw2t                  ^libxrender@0.9.10%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  pqxzv26                      ^renderproto@0.11.1%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  nmzycnm              ^libxscrnsaver@1.2.2%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  nj6glps                  ^scrnsaverproto@1.2.2%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  simblvj      ^cmake@3.26.3%gcc@11.3.0~doc+ncurses+ownlibs~qt build_system=generic build_type=Release arch=linux-rhel8-zen2
[+]  k6m5lie          ^ncurses@6.4%gcc@11.3.0+symlinks+termlib abi=5 build_system=autotools arch=linux-rhel8-zen2
[+]  bjvjttn          ^openssl@1.1.1v%gcc@11.3.0~docs+shared build_system=generic certs=system arch=linux-rhel8-zen2
[+]  ohlruz3      ^faodel@1.2108.1%gcc@11.3.0+hdf5+ipo+mpi+shared+tcmalloc build_system=cmake build_type=RelWithDebInfo generator=make logging=stdout network=nnti serializer=xdr arch=linux-rhel8-zen2
[+]  dk7icze          ^boost@1.82.0%gcc@11.3.0+atomic+chrono~clanglibcpp+container+context+contract+coroutine+date_time~debug+exception+fiber+filesystem+graph+graph_parallel+icu+iostreams+json+locale+log+math+mpi+multithreaded+nowide+numpy+pic+program_options+python+random+regex+serialization+shared+signals~singlethreaded+stacktrace+system~taggedlayout+test+thread+timer+type_erasure~versionedlayout+wave build_system=generic context-impl=fcontext cxxstd=17 patches=a440f96,a7c807f,b8569d7 visibility=hidden arch=linux-rhel8-zen2
[+]  onmsrte              ^icu4c@67.1%gcc@11.3.0 build_system=autotools cxxstd=17 arch=linux-rhel8-zen2
[+]  zfccli6              ^py-numpy@1.24.3%gcc@11.3.0+blas+lapack build_system=python_pip patches=873745d arch=linux-rhel8-zen2
[+]  prjehhd                  ^openblas@0.3.23%gcc@11.3.0~bignuma~consistent_fpcsr+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=openmp arch=linux-rhel8-zen2
[+]  pkcpzcq                  ^py-cython@0.29.33%gcc@11.3.0 build_system=python_pip patches=71de066 arch=linux-rhel8-zen2
[+]  3juduhn                  ^py-pip@23.0%gcc@11.3.0 build_system=generic arch=linux-rhel8-zen2
[+]  63t3joz                  ^py-setuptools@63.4.3%gcc@11.3.0 build_system=generic arch=linux-rhel8-zen2
[+]  yiqallw                  ^py-wheel@0.37.1%gcc@11.3.0 build_system=generic arch=linux-rhel8-zen2
[+]  qoumw5v              ^python@3.10.10%gcc@11.3.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma+nis+optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,7d40923,f2fd060 arch=linux-rhel8-zen2
[+]  5wi2hwp                  ^libnsl@2.0.0%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  bgjnz6u                      ^rpcsvc-proto@1.4.3%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  5qlirxs                  ^sqlite@3.40.1%gcc@11.3.0+column_metadata+dynamic_extensions+fts+functions+rtree build_system=autotools arch=linux-rhel8-zen2
[+]  ranwi3u              ^xz@5.4.1%gcc@11.3.0+pic build_system=autotools libs=shared,static arch=linux-rhel8-zen2
[+]  dyhotaa      ^fmt@10.0.0%gcc@11.3.0+ipo+pic+shared build_system=cmake build_type=RelWithDebInfo cxxstd=17 generator=make arch=linux-rhel8-zen2
[+]  bcadp5n      ^gmake@4.4.1%gcc@11.3.0+guile build_system=autotools arch=linux-rhel8-zen2
[+]  fny73jv          ^guile@2.2.6%gcc@11.3.0+readline build_system=autotools threads=none arch=linux-rhel8-zen2
[+]  zy5n3jc              ^bdw-gc@8.2.2%gcc@11.3.0+libatomic-ops build_system=autotools threads=none arch=linux-rhel8-zen2
[+]  tkxbqtj                  ^libatomic-ops@7.8.0%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  aym4bs5              ^gettext@0.21.1%gcc@11.3.0+bzip2+curses+git+libunistring+libxml2+tar+xz build_system=autotools arch=linux-rhel8-zen2
[+]  vjlzfma                  ^tar@1.34%gcc@11.3.0 build_system=autotools zip=pigz arch=linux-rhel8-zen2
[+]  pnsotvl                      ^pigz@2.7%gcc@11.3.0 build_system=makefile arch=linux-rhel8-zen2
[+]  rbqwy2o              ^gmp@6.2.1%gcc@11.3.0+cxx build_system=autotools libs=shared,static patches=69ad2e2 arch=linux-rhel8-zen2
[+]  akxkimj              ^libffi@3.4.4%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  ovq4gsp              ^libtool@2.4.7%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  zra6tcb              ^libunistring@1.1%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  boheg7m              ^readline@8.2%gcc@11.3.0 build_system=autotools patches=bbf97f1 arch=linux-rhel8-zen2
[+]  wqtnva5          ^pkg-config@0.29.2%gcc@11.3.0+internal_glib build_system=autotools arch=linux-rhel8-zen2
[+]  7ghnnez      ^libx11@1.8.4%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  btqs4tq          ^inputproto@2.3.2%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  byzz5dp          ^kbproto@1.0.7%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  ow2zhuk          ^libxcb@1.14%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  mtrhc3f              ^libpthread-stubs@0.4%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  tfumrao              ^libxau@1.0.8%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  66aqrtf              ^libxdmcp@1.1.4%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  h5im2xa                  ^libbsd@0.11.7%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  5sslvdw                      ^libmd@1.0.4%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  nb5istf              ^xcb-proto@1.15.2%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  sve5wjh          ^perl@5.36.0%gcc@11.3.0+cpanm+open+shared+threads build_system=generic arch=linux-rhel8-zen2
[+]  nayr6pc              ^berkeley-db@18.1.40%gcc@11.3.0+cxx+docs+stl build_system=autotools patches=26090f4 arch=linux-rhel8-zen2
[+]  tspio6g              ^gdbm@1.23%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  5s2rtxx          ^util-macros@1.19.3%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  hk7ai2i          ^xextproto@7.3.0%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  wmblrok          ^xproto@7.0.31%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  nsk6mdj          ^xtrans@1.4.0%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  2h7squc      ^matio@1.5.17%gcc@11.3.0+hdf5+shared+zlib build_system=autotools arch=linux-rhel8-zen2
[+]  gv2fyi6          ^zlib@1.2.13%gcc@11.3.0+optimize+pic+shared build_system=makefile arch=linux-rhel8-zen2
[+]  v7nbj47      ^netcdf-c@4.9.2%gcc@11.3.0+blosc+byterange+dap~fsync+hdf4+jna+mpi+nczarr_zip+optimize+parallel-netcdf+pic+shared+szip+zstd build_system=autotools arch=linux-rhel8-zen2
[+]  y2p2gbz          ^bzip2@1.0.8%gcc@11.3.0~debug+pic+shared build_system=generic arch=linux-rhel8-zen2
[+]  nbfheoo              ^diffutils@3.9%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  xvfrkmu          ^c-blosc@1.21.2%gcc@11.3.0+avx2+ipo build_system=cmake build_type=RelWithDebInfo generator=make arch=linux-rhel8-zen2
[+]  qmkbina              ^lz4@1.9.4%gcc@11.3.0 build_system=makefile libs=shared,static arch=linux-rhel8-zen2
[+]  ifpmjvp              ^snappy@1.1.10%gcc@11.3.0+ipo+pic+shared build_system=cmake build_type=RelWithDebInfo generator=make arch=linux-rhel8-zen2
[+]  ml7o4bu          ^curl@7.88.1%gcc@11.3.0+gssapi~ldap+libidn2~librtmp+libssh+libssh2+nghttp2 build_system=autotools libs=shared,static tls=gnutls,mbedtls,openssl arch=linux-rhel8-zen2
[+]  vvo2utu              ^gnutls@3.7.8%gcc@11.3.0+guile+zlib build_system=autotools arch=linux-rhel8-zen2
[+]  zksb6q3                  ^nettle@3.8.1%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  hoxnvw7              ^krb5@1.20.1%gcc@11.3.0+shared build_system=autotools arch=linux-rhel8-zen2
[+]  z4v5kwp              ^libidn2@2.3.4%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  5et2bgd              ^libssh@0.8.9%gcc@11.3.0+gssapi+ipo build_system=cmake build_type=RelWithDebInfo generator=make arch=linux-rhel8-zen2
[+]  glfk6ep              ^libssh2@1.10.0%gcc@11.3.0+ipo+shared build_system=cmake build_type=RelWithDebInfo crypto=mbedtls generator=make arch=linux-rhel8-zen2
[+]  ek4c3lz              ^mbedtls@2.28.1%gcc@11.3.0+pic build_system=makefile build_type=RelWithDebInfo libs=shared,static arch=linux-rhel8-zen2
[+]  ddcqjdi              ^nghttp2@1.52.0%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  ir36lb6          ^hdf@4.2.15%gcc@11.3.0+external-xdr+fortran~java~netcdf+pic~shared+szip build_system=autotools patches=49733dd,5434f29,ce75518,d54592d,f7d67e8 arch=linux-rhel8-zen2
[+]  idcnjva              ^bison@3.8.2%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  22ydxmr              ^flex@2.6.4%gcc@11.3.0+lex+nls build_system=autotools patches=f8b85a0 arch=linux-rhel8-zen2
[+]  2wuk76j                  ^findutils@4.9.0%gcc@11.3.0 build_system=autotools patches=440b954 arch=linux-rhel8-zen2
[+]  pw6daqo                  ^help2man@1.49.3%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  qbo23df              ^libjpeg-turbo@2.1.5%gcc@11.3.0+ipo~jpeg8+shared+static build_system=cmake build_type=RelWithDebInfo generator=make arch=linux-rhel8-zen2
[+]  rgjx3pn                  ^nasm@2.15.05%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  x3keyoo              ^libtirpc@1.2.6%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  zyvinkw          ^libszip@2.1.1%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  hfk7szf          ^libxml2@2.10.3%gcc@11.3.0~python build_system=autotools arch=linux-rhel8-zen2
[+]  ghqoznx              ^libiconv@1.17%gcc@11.3.0 build_system=autotools libs=shared,static arch=linux-rhel8-zen2
[+]  36epzo4          ^libzip@1.3.2%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  zbqok4x          ^parallel-netcdf@1.12.3%gcc@11.3.0~burstbuffer+cxx+fortran+pic+shared build_system=autotools arch=linux-rhel8-zen2
[+]  2xgh4yr              ^m4@1.4.19%gcc@11.3.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-rhel8-zen2
[+]  lskfqtb                  ^libsigsegv@2.14%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  5hgslof          ^zstd@1.5.5%gcc@11.3.0+programs build_system=makefile compression=lz4,lzma,zlib libs=shared,static arch=linux-rhel8-zen2
[+]  zw2zshc      ^openmpi@4.1.5%gcc@11.3.0+atomics~cuda+cxx+cxx_exceptions~gpfs~internal-hwloc~java+legacylaunchers~lustre~memchecker+orterunprefix+pmi+romio+rsh~singularity~static~vt+wrapper-rpath build_system=autotools fabrics=ucx schedulers=slurm arch=linux-rhel8-zen2
[+]  hyze5je          ^hwloc@2.9.1%gcc@11.3.0~cairo~cuda~gl+libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-rhel8-zen2
[+]  qga6koi              ^libpciaccess@0.17%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  262ta3t          ^numactl@2.0.14%gcc@11.3.0 build_system=autotools patches=4e1d78c,62fc8a8,ff37630 arch=linux-rhel8-zen2
[+]  yxpydi6              ^autoconf@2.69%gcc@11.3.0 build_system=autotools patches=35c4492,7793209,a49dd5b arch=linux-rhel8-zen2
[+]  vr5d2na              ^automake@1.16.5%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  r4hghfg          ^openssh@9.3p1%gcc@11.3.0+gssapi build_system=autotools arch=linux-rhel8-zen2
[+]  egb5eml              ^libedit@3.1-20210216%gcc@11.3.0 build_system=autotools arch=linux-rhel8-zen2
[+]  5zyvitn              ^libxcrypt@4.4.33%gcc@11.3.0~obsolete_api build_system=autotools arch=linux-rhel8-zen2
[+]  vxjuhaw          ^pmix@3.2.3%gcc@11.3.0~docs+pmi_backwards_compatibility build_system=autotools arch=linux-rhel8-zen2
[+]  l3smejv          ^slurm@system%gcc@11.3.0~gtk~hdf5~hwloc~mariadb~pmix+readline~restd build_system=autotools sysconfdir=/etc/slurm arch=linux-rhel8-zen2
[+]  ktl3nrn          ^ucx@1.15.0%gcc@11.3.0~assertions~backtrace_detail+cma~cuda~dc~debug~dm+examples~gdrcopy~gtest~ib_hw_tm~java+knem~logging~mlx5_dv+openmp+optimizations~parameter_checking+pic~rc+rdmacm~rocm+thread_multiple~ucg~ud~verbs~vfs+xpmem build_system=autotools libs=shared opt=3 simd=auto arch=linux-rhel8-zen2
```


### Error message

<details><summary>Error message</summary>

<pre>
...
-- [Faodel] configured, available targets in /cvmfs/hpcsw.umd.edu/spack-software/2023.05.30/linux-rhel8-zen2/gcc-11.3.0/faodel-1.2108.1-ohlruz3zj53gz34pbtxyda2iyiqtbsnu/lib/cmake/Faodel/FaodelTargets.cmake
-- Faodel_LIBRARY_NAMES='kelpie;dirman;opbox;lunasa;nnti;faodel-services;whookie;faodel-common;sbl;tcmalloc;spinlock'
-- TPL_Faodel_LIBRARIES='Faodel::kelpie'
-- TPL_Faodel_INCLUDE_DIRS='/cvmfs/hpcsw.umd.edu/spack-software/2023.05.30/linux-rhel8-zen2/gcc-11.3.0/faodel-1.2108.1-ohlruz3zj53gz34pbtxyda2iyiqtbsnu/include'
CMake Error at cmake/tribits/core/utils/MessageWrapper.cmake:77 (message):
  ERROR: Can't handle argument 'Faodel::kelpie' in list TPL_Faodel_LIBRARIES
Call Stack (most recent call first):
  cmake/tribits/core/package_arch/TribitsExternalPackageWriteConfigFile.cmake:540 (message_wrapper)
  cmake/tribits/core/package_arch/TribitsExternalPackageWriteConfigFile.cmake:402 (tribits_extpkg_process_libraries_list)
  cmake/tribits/core/package_arch/TribitsExternalPackageWriteConfigFile.cmake:82 (tribits_extpkg_write_config_file_str)
  cmake/tribits/core/package_arch/TribitsTplFindIncludeDirsAndLibraries.cmake:724 (tribits_extpkg_write_config_file)
  packages/seacas/cmake/tpls/FindTPLFaodel.cmake:100 (TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES)
  cmake/tribits/core/package_arch/TribitsProcessEnabledTpls.cmake:189 (include)
  cmake/tribits/core/package_arch/TribitsProcessEnabledTpls.cmake:139 (tribits_process_enabled_tribits_find_tpl_mod_file)
  cmake/tribits/core/package_arch/TribitsProcessEnabledTpls.cmake:95 (tribits_process_enabled_standard_tpl)
  cmake/tribits/core/package_arch/TribitsProjectImpl.cmake:193 (tribits_process_enabled_tpls)
  cmake/tribits/core/package_arch/TribitsProject.cmake:92 (tribits_project_impl)
  CMakeLists.txt:46 (TRIBITS_PROJECT)
</pre></details>


### Information on your system

```console
spack debug report
* **Spack:** 0.21.0.dev0 (84faf5a6cf342efddb976b7432e44aff12f1c985)
* **Python:** 3.6.8
* **Platform:** linux-rhel8-zen2
* **Concretizer:** clingo
```

but with the updated seacas recipe from #39756 (including support for @2023-05-30 and fmt@10)

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/12752108/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/12752109/spack-build-out.txt)

Maintainers: @gsjaardema

It looks like seacas made some major changes in the Cmake stuff for finding faodel with commit c65d7547 in Feb 2021, and I think as a result we need to define Cmake variables Faodel_INCLUDE_DIRS and Faodel_LIBRARY_DIRS now. 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/seacas/package.py
+++ b/var/spack/repos/builtin/packages/seacas/package.py
@@ -494,7 +494,7 @@ class Seacas(CMakePackage):
             # leaves off the faodel part
             faodel_incdir = spec["faodel"].prefix.include
             faodel_incdir2 = spec["faodel"].prefix.include.faodel
-            faodel_incdirs = [ faodel_incdir, faodel_incdir2 ]
+            faodel_incdirs = [faodel_incdir, faodel_incdir2]
             options.append(define("Faodel_INCLUDE_DIRS", ";".join(faodel_incdirs)))
             options.append(define("Faodel_LIBRARY_DIRS", spec["faodel"].prefix.lib))
 

--- a/var/spack/repos/builtin/packages/seacas/package.py
+++ b/var/spack/repos/builtin/packages/seacas/package.py
@@ -304,6 +304,9 @@ class Seacas(CMakePackage):
         when="@:2023-10-24",
     )
 
+    # Based on install-tpl.sh script, cereal seems to only be used when faodel enabled
+    depends_on("cereal", when="@2021-04-02: +faodel")
+
     def setup_run_environment(self, env):
         env.prepend_path("PYTHONPATH", self.prefix.lib)
 
@@ -485,6 +488,15 @@ class Seacas(CMakePackage):
         for pkg in ("Faodel", "BOOST"):
             if pkg.lower() in spec:
                 options.append(define(pkg + "_ROOT", spec[pkg.lower()].prefix))
+
+        if "+faodel" in spec:
+            # faodel headers are under $faodel_prefix/include/faodel but seacas
+            # leaves off the faodel part
+            faodel_incdir = spec["faodel"].prefix.include
+            faodel_incdir2 = spec["faodel"].prefix.include.faodel
+            faodel_incdirs = [ faodel_incdir, faodel_incdir2 ]
+            options.append(define("Faodel_INCLUDE_DIRS", ";".join(faodel_incdirs)))
+            options.append(define("Faodel_LIBRARY_DIRS", spec["faodel"].prefix.lib))
 
         options.append(from_variant("TPL_ENABLE_ADIOS2", "adios2"))
         if "+adios2" in spec:

